src/components/editMeasure/testCases/components/testCaseLanding/common/ActionCenter/ActionCenter.test.tsx

```diff
@@ -468,5 +468,232 @@ describe("ActionCenter Component", () => {
+            <ActionCenter
+              selectedTestCases={[
+                { id: "1", validResource: true, title: "Test Case 1" },
+              ]}
+              canEdit={false}
+              isQDM={false}
+              isDraft={true}
+            />
+          </MemoryRouter>
+        );
+
+        expect(
+          screen.queryByTestId("make-json-match-ui-action-btn")
+        ).not.toBeInTheDocument();
+      });
+
+      it("should not display Make JSON Match UI button for QDM measures", () => {
+        render(
+          <MemoryRouter>
+            <ActionCenter
+              selectedTestCases={[
+                { id: "1", validResource: true, title: "Test Case 1" },
+              ]}
+              canEdit={true}
+              isQDM={true}
+              isDraft={true}
+            />
+          </MemoryRouter>
+        );
+
+        expect(
+          screen.queryByTestId("make-json-match-ui-action-btn")
+        ).not.toBeInTheDocument();
+      });
+
+      it("should display Make JSON Match UI button for QI-Core measures when feature flag is enabled and user has edit access", () => {
+        render(
+          <MemoryRouter>
+            <ActionCenter
+              selectedTestCases={[
+                { id: "1", validResource: true, title: "Test Case 1" },
+              ]}
+              canEdit={true}
+              isQDM={false}
+              isDraft={true}
+            />
+          </MemoryRouter>
+        );
+
+        expect(
+          screen.getByTestId("make-json-match-ui-action-btn")
+        ).toBeInTheDocument();
+      });
+
+      it("should enable Make JSON Match UI button when one test case is selected", () => {
+        render(
+          <MemoryRouter>
+            <ActionCenter
+              selectedTestCases={[
+                { id: "1", validResource: true, title: "Test Case 1" },
+              ]}
+              canEdit={true}
+              isQDM={false}
+              isDraft={true}
+            />
+          </MemoryRouter>
+        );
+
+        const makeJsonMatchUiBtn = screen.getByTestId(
+          "make-json-match-ui-action-btn"
+        );
+        expect(makeJsonMatchUiBtn).toBeEnabled();
+      });
+
+      it("should enable Make JSON Match UI button when multiple test cases are selected", () => {
+        render(
+          <MemoryRouter>
+            <ActionCenter
+              selectedTestCases={[
+                { id: "1", validResource: true, title: "Test Case 1" },
+                { id: "2", validResource: true, title: "Test Case 2" },
+              ]}
+              canEdit={true}
+              isQDM={false}
+              isDraft={true}
+            />
+          </MemoryRouter>
+        );
+
+        const makeJsonMatchUiBtn = screen.getByTestId(
+          "make-json-match-ui-action-btn"
+        );
+        expect(makeJsonMatchUiBtn).toBeEnabled();
+      });
+
+      it("should disable Make JSON Match UI button when no test cases are selected", () => {
```

Comment:
same here- this test case and the following can be combined. just a difference in one assertion.